src/util/pointingabilities.h

```diff
@@ -0,0 +1,62 @@
+/*
+Minetest
+Copyright (C) 2010-2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#pragma once
+#include <string>
+#include <unordered_map>
+#include "itemgroup.h"
+#include <optional>
+
+enum PointabilityType
+{
+	POINTABLE,
+	POINTABLE_NOT, // Can be pointed through.
+	POINTABLE_BLOCKING,
+};
+
+// An object to store overridden pointable properties
+struct PointingAbilities
```

Comment:
Actually point-ability is wrong (even though it is shorter and more convenient), since the ability does not refer to the noun "point" but to the verb "to point", so it should be "pointing abilities".
I changed it in the lua api for consistency, and renamed PointabilityType to PointableType.

Also, I though about making it singular (ability instead of abilities), but Imo plural makes more sense, since it refers to the pointing ability of multiple items/objects, and tool capabilities is also plural.

Reply:
Actually point-ability is wrong, since the ability does not refer to the noun "point" but to the verb "to point", but I guess it is OK since it's shorter and more convenient.

I changed it to `pointabilities` like you suggested. It is now consistent with the lua api.

Also, I though about making it singular (ability instead of abilities), but Imo plural makes more sense, since it refers to the pointing ability of multiple items/objects, and tool capabilities is also plural.

(I'm bad at naming things.)